### PR TITLE
feat: bring disk encryption support

### PR DESCRIPTION
**Pull request by smira**
Mostly copy-paste from `v1` of `go-blockdevice`, but with some fixes to adapt to new code in other places.

### CODE REVIEW

**Review by frezbo — APPROVED** [review]

File: encryption/luks/token.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// Token defines LUKS2 token.
+type Token[UserData any] struct {
+	UserData UserData `json:",inline"`
+	Type     string   `json:"type"`
+}
+
+// Bytes encodes token into bytes.
+func (t *Token[UserData]) Bytes() ([]byte, error) {
+	return json.Marshal(struct {
+		*Token[UserData] `json:",inline"`
```
Comment (DmitriyMV) [inline encryption/luks/token.go:18]: There is no `inline` in "encoding/json".

File: encryption/luks/token.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package luks
+
+import "encoding/json"
+
+// Token defines LUKS2 token.
+type Token[UserData any] struct {
+	UserData UserData `json:",inline"`
```
Comment (DmitriyMV) [inline encryption/luks/token.go:11]: There is no `inline` in "encoding/json".

File: encryption/luks/luks.go [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+	args := []string{
+		"luksAddKey",
+		devname,
+		"--key-file=-",
+		fmt.Sprintf("--keyfile-size=%d", keyfileLen),
+	}
+
+	args = append(args, l.argonArgs()...)
+	args = append(args, l.encryptionArgs()...)
+	args = append(args, keyslotArgs(newKey)...)
```
Comment (DmitriyMV) [inline encryption/luks/luks.go:201]:
```suggestion
	args := slices.Concat(
		[]string{
			"luksAddKey",
			devname,
			"--key-file=-",
			fmt.Sprintf("--keyfile-size=%d", keyfileLen),
		},
		l.argonArgs(),
		l.encryptionArgs(),
		keyslotArgs(newKey),
	)
```

File: encryption/luks/luks.go [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+func (l *LUKS) Encrypt(deviceName string, key *encryption.Key) error {
+	cipher, err := l.cipher.String()
+	if err != nil {
+		return err
+	}
+
+	args := []string{"luksFormat", "--type", "luks2", "--key-file=-", "-c", cipher, deviceName}
+	args = append(args, l.argonArgs()...)
+	args = append(args, keyslotArgs(key)...)
+	args = append(args, l.encryptionArgs()...)
```
Comment (DmitriyMV) [inline encryption/luks/luks.go:158]:
```suggestion
	args := slices.Concat(
		[]string{"luksFormat", "--type", "luks2", "--key-file=-", "-c", cipher, deviceName},
		l.argonArgs(),
		keyslotArgs(key),
		l.encryptionArgs(),
	)
```

File: encryption/luks/luks.go [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+	if l.keySize == 0 {
+		l.keySize = keySizeDefaults[cipher]
+	}
+
+	return l
+}
+
+// Open runs luksOpen on a device and returns mapped device path.
+func (l *LUKS) Open(deviceName, mappedName string, key *encryption.Key) (string, error) {
+	args := []string{"luksOpen", deviceName, mappedName, "--key-file=-"}
```
Comment (DmitriyMV) [inline encryption/luks/luks.go:136]:
```
slices.Concat 
```

**Comment by smira** [comment]
thanks, fixed up all of that

**Review by Unix4ever — APPROVED** [review]

File: encryption/luks/token.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package luks
+
+import "encoding/json"
+
+// Token defines LUKS2 token.
+type Token[UserData any] struct {
+	// UserData has a strange JSON tag, but this keeps it backwards compatible with v1 library.
```
Comment (smira) [inline encryption/luks/token.go:11]: added some clarity and tests here

**Review by frezbo — APPROVED** [review]

**Pull request merged**